src/display/touch_manager.js

```diff
@@ -0,0 +1,195 @@
+/* Copyright 2024 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { shadow } from "../shared/util.js";
+
+class TouchManager {
+  #container;
+
+  #isPinching = false;
+
+  #isPinchingDisabled;
+
+  #overlayManager;
+
+  #onPinching;
+
+  #onPinchEnd;
+
+  #signal;
+
+  #touchInfo = null;
+
+  #touchManagerAC;
+
+  #touchMoveAC = null;
+
+  static #MIN_TOUCH_DISTANCE_TO_PINCH = 0;
+
+  constructor({
+    container,
+    overlayManager = null,
```

Comment:
Rather than having to pass in an `OverlayManager`-instance, how about adding another callback function *similar* to the one just below?